src/main.nr

```diff
@@ -7,14 +9,30 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
```

Comment:
You can try doing something like this to avoid having to do the % operation, so that Noir will take care of that itself.

```
fn calculate_hash(num: u32) -> u32 {
    let num_field = num as Field;
    let num_bytes = num_field.to_be_bytes(4);
    let size = 2.pow_32(32) as u64;
    let mut hash_value: u32 = 0x811C9DC5;

    for i in 0..4 {
        let product = hash_value * PRIME;
        hash_value = (product ^ (num_bytes[i] as u32));
    }

    hash_value
}
```